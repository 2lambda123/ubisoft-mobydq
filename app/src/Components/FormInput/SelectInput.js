import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormHelperText from '@material-ui/core/FormHelperText';
import Select from '@material-ui/core/Select';

const SelectInput = ({
  id,
  label,
  helperText,
  error,
  touched,
  errors,
  value,
  onChange,
  className,
  ...props
}) => {
  return (
    <Select
      id={id}
      name={id}
      value={value}
      onChange={onChange}
    >
      <MenuItem value="">
        <em>None</em>
      </MenuItem>
      <MenuItem value={10}>Ten</MenuItem>
      <MenuItem value={20}>Twenty</MenuItem>
      <MenuItem value={30}>Thirty</MenuItem>
  </Select>
  );
};

export default SelectInput;


SelectInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};