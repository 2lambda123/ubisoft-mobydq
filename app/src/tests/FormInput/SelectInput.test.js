import React from 'react'
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SelectInput from '../../Components/FormInput/SelectInput';

describe('TextInput unit test', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SelectInput
        id="Id"
        label="Label"
        items={[{id: 1, name: 'select value 1'}, {id: 2, name: 'select value 2'}]}
        helperText="Helper Text"
        placeholder="Enter indicator name"
        touched={jest.fn()}
        error=""
        value="value"
        onChange={jest.fn()}
        onBlur={jest.fn()}
      />
    );
  });

  it('renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('TextInput functional test', () => {
  let wrapper = mount(
      <SelectInput
        id="Id"
        label="Label"
        items={[{id: 1, name: 'select value 1'}, {id: 2, name: 'select value 2'}]}
        helperText="Helper Text"
        placeholder="Enter indicator name"
        touched={jest.fn()}
        error=""
        value={1}
        onChange={jest.fn()}
        onBlur={jest.fn()}
      />
    );
  it ('renders correct contents', () => {
    expect(wrapper.prop('value')).toEqual(1)
  });
});
