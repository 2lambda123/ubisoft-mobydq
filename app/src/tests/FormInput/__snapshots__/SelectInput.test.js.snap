// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectInput
    error=""
    helperText="Helper Text"
    id="Id"
    items={
      Array [
        Object {
          "id": 1,
          "name": "select value 1",
        },
        Object {
          "id": 2,
          "name": "select value 2",
        },
      ]
    }
    label="Label"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder="Select indicator type"
    touched="true"
    value="value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(InputLabel)
          htmlFor="Id"
        >
          Label
        </WithStyles(InputLabel)>,
        <WithStyles(Select)
          helperText="Helper Text"
          id="Id"
          inputProps={
            Object {
              "id": Object {
                "id": "Id",
              },
            }
          }
          name="Id"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          placeholder="Select indicator type"
          value="value"
        >
          <WithStyles(MenuItem)
            value=""
          >
            <em>
              Select..
            </em>
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
            value={1}
          >
            select value 1
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
            value={2}
          >
            select value 2
          </WithStyles(MenuItem)>
        </WithStyles(Select)>,
        <WithStyles(FormHelperText)>
          
        </WithStyles(FormHelperText)>,
      ],
      "error": false,
      "style": Object {
        "margin": "15px",
        "marginLeft": "0px",
        "width": "240px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Label",
          "htmlFor": "Id",
        },
        "ref": null,
        "rendered": "Label",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(MenuItem)
              value=""
            >
              <em>
                Select..
              </em>
            </WithStyles(MenuItem)>,
            Array [
              <WithStyles(MenuItem)
                value={1}
              >
                select value 1
              </WithStyles(MenuItem)>,
              <WithStyles(MenuItem)
                value={2}
              >
                select value 2
              </WithStyles(MenuItem)>,
            ],
          ],
          "helperText": "Helper Text",
          "id": "Id",
          "inputProps": Object {
            "id": Object {
              "id": "Id",
            },
          },
          "name": "Id",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "placeholder": "Select indicator type",
          "value": "value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <em>
                Select..
              </em>,
              "value": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select..",
              },
              "ref": null,
              "rendered": "Select..",
              "type": "em",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": "select value 1",
              "value": 1,
            },
            "ref": null,
            "rendered": "select value 1",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "children": "select value 2",
              "value": 2,
            },
            "ref": null,
            "rendered": "select value 2",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(InputLabel)
            htmlFor="Id"
          >
            Label
          </WithStyles(InputLabel)>,
          <WithStyles(Select)
            helperText="Helper Text"
            id="Id"
            inputProps={
              Object {
                "id": Object {
                  "id": "Id",
                },
              }
            }
            name="Id"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            placeholder="Select indicator type"
            value="value"
          >
            <WithStyles(MenuItem)
              value=""
            >
              <em>
                Select..
              </em>
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              value={1}
            >
              select value 1
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              value={2}
            >
              select value 2
            </WithStyles(MenuItem)>
          </WithStyles(Select)>,
          <WithStyles(FormHelperText)>
            
          </WithStyles(FormHelperText)>,
        ],
        "error": false,
        "style": Object {
          "margin": "15px",
          "marginLeft": "0px",
          "width": "240px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Label",
            "htmlFor": "Id",
          },
          "ref": null,
          "rendered": "Label",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(MenuItem)
                value=""
              >
                <em>
                  Select..
                </em>
              </WithStyles(MenuItem)>,
              Array [
                <WithStyles(MenuItem)
                  value={1}
                >
                  select value 1
                </WithStyles(MenuItem)>,
                <WithStyles(MenuItem)
                  value={2}
                >
                  select value 2
                </WithStyles(MenuItem)>,
              ],
            ],
            "helperText": "Helper Text",
            "id": "Id",
            "inputProps": Object {
              "id": Object {
                "id": "Id",
              },
            },
            "name": "Id",
            "onBlur": [MockFunction],
            "onChange": [MockFunction],
            "placeholder": "Select indicator type",
            "value": "value",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <em>
                  Select..
                </em>,
                "value": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select..",
                },
                "ref": null,
                "rendered": "Select..",
                "type": "em",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": "select value 1",
                "value": 1,
              },
              "ref": null,
              "rendered": "select value 1",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "children": "select value 2",
                "value": 2,
              },
              "ref": null,
              "rendered": "select value 2",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
